### PR TITLE
Add support for lambda syntax that allows to specify parameter name. …

### PR DESCRIPTION
**Pull request by NazarKacharaba**
Fix doc generation for lambdas.

### CODE REVIEW

File: gluecodium/src/main/resources/templates/cpp/CppLambdaDoc.mustache [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{{!!
+  !
+  ! Copyright (C) 2016-2023 HERE Europe B.V.
```
Comment (sameermahant) [inline gluecodium/src/main/resources/templates/cpp/CppLambdaDoc.mustache:3]:
```suggestion
  ! Copyright (C) 2023 HERE Europe B.V.
```

File: gluecodium/src/test/resources/smoke/comments/input/CommentsLambda.lime [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  // The first line of the doc.
+  // @param[inputParameter] The first input parameter. The second sentence of the first input parameter.
+  // @return The string.
+  lambda WithNamedParameters = (inputParameter: String) -> String
+
+  // The first line of the doc.
+  // @param[p0] The first input parameter.
+  // @return The string.
+  lambda MixedDocNameParameters = (inputParameter: String, secondInputParameter: String) -> String
```
Comment (sameermahant) [inline gluecodium/src/test/resources/smoke/comments/input/CommentsLambda.lime:37]: Can we prevent such mixed usage?

File: gluecodium/src/main/resources/templates/dart/DartLambdaDocs.mustache [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{{!!
+  !
+  ! Copyright (C) 2016-2023 HERE Europe B.V.
```
Comment (sameermahant) [inline gluecodium/src/main/resources/templates/dart/DartLambdaDocs.mustache:3]:
```suggestion
  ! Copyright (C) 2023 HERE Europe B.V.
```

File: gluecodium/src/main/resources/templates/swift/SwiftLambdaComment.mustache [hunk 4]
```diff
@@ -0,0 +1,33 @@
+{{!!
+  !
+  ! Copyright (C) 2016-2023 HERE Europe B.V.
```
Comment (sameermahant) [inline gluecodium/src/main/resources/templates/swift/SwiftLambdaComment.mustache:3]:
```suggestion
  ! Copyright (C) 2023 HERE Europe B.V.
```

File: lime-runtime/src/main/java/com/here/gluecodium/model/lime/LimeLambdaParameter.kt [hunk 5]
```diff
@@ -20,10 +20,16 @@
...
+    comment: LimeComment = LimeComment(),
+    attributes: LimeAttributes? = null,
+    @Suppress("unused") val isNamedParameter: Boolean = false
+) : LimeTypedElement(path, comment, attributes, typeRef = typeRef) {
+
+    fun asLimeParameter() =
         LimeParameter(path = path, comment = comment, attributes = attributes, typeRef = typeRef)
+
+    @Suppress("unused")
+    fun parameterName() = path.name
```
Comment (sameermahant) [inline lime-runtime/src/main/java/com/here/gluecodium/model/lime/LimeLambdaParameter.kt:34]: Needed?

File: lime-runtime/src/main/java/com/here/gluecodium/model/lime/LimeLambdaParameter.kt [hunk 6]
```diff
@@ -20,10 +20,16 @@
...
-    val typeRef: LimeTypeRef,
-    val comment: LimeComment = LimeComment(),
-    attributes: LimeAttributes? = null
-) : LimeElement(attributes) {
-    fun asParameter(path: LimePath) =
+    typeRef: LimeTypeRef,
+    path: LimePath,
+    comment: LimeComment = LimeComment(),
+    attributes: LimeAttributes? = null,
+    @Suppress("unused") val isNamedParameter: Boolean = false
```
[the remark this answers is not in the document]
Comment (NazarKacharaba) [reply lime-runtime/src/main/java/com/here/gluecodium/model/lime/LimeLambdaParameter.kt:27]: We use Suppress here as Kotlin does not know that it is used in the mustache template - https://github.com/heremaps/gluecodium/pull/1546/files#diff-cc7abef0c5c0015c2dfe268757dff7c7566d6462db541fdd797ce7cf6e1de384R23

File: lime-runtime/src/main/java/com/here/gluecodium/model/lime/LimeLambdaParameter.kt (hunk 5 above)
Comment (NazarKacharaba) [reply lime-runtime/src/main/java/com/here/gluecodium/model/lime/LimeLambdaParameter.kt:34]: Same here, it is used in mustache template files.

File: gluecodium/src/test/resources/smoke/comments/input/CommentsLambda.lime (hunk 2 above)
Comment (NazarKacharaba) [reply gluecodium/src/test/resources/smoke/comments/input/CommentsLambda.lime:37]: It is aligned with the behavior for the doc functions: if the parameter name in the doc does not match with one in the lambda definition, then this parameter will not be included in the documentation.

File: gluecodium/src/main/resources/templates/swift/SwiftLambdaComment.mustache (hunk 4 above)
Comment (NazarKacharaba) [reply gluecodium/src/main/resources/templates/swift/SwiftLambdaComment.mustache:3]: This string is generated using `./gradlew :gluecodium:spotlessApply`, `build` will fail if we do not apply the spotless check, see - https://github.com/heremaps/gluecodium/blob/master/config/spotless/here_mustache.license#L3

File: gluecodium/src/main/resources/templates/cpp/CppLambdaDoc.mustache (hunk 1 above)
Comment (NazarKacharaba) [reply gluecodium/src/main/resources/templates/cpp/CppLambdaDoc.mustache:3]: This string is generated using `./gradlew :gluecodium:spotlessApply`, `build` will fail if we do not apply the spotless check, see - https://github.com/heremaps/gluecodium/blob/master/config/spotless/here_mustache.license#L3

**Review by sameermahant — APPROVED** [review]

**Pull request merged**